Editor/Utility/CinemachineEditorAnalytics.cs

```diff
@@ -0,0 +1,251 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace Cinemachine.Editor
+{
+    [InitializeOnLoad]
+    static class CinemachineEditorAnalytics
+    {
+        const int k_MaxEventsPerHour = 1000;
+        const int k_MaxNumberOfElements = 1000;
+        const string k_VendorKey = "unity.cinemachine";
+
+        static List<Type> s_CinemachineAssemblyTypes;
+
+        // register an event handler when the class is initialized
+        static CinemachineEditorAnalytics()
+        {
+            EditorApplication.playModeStateChanged += SendAnalyticsOnPlayEnter;
+            
+            // Query for all types in cinemachine assembly
+            s_CinemachineAssemblyTypes = typeof(CinemachineBrain).Assembly.GetTypes().
+                Where(t => t.IsClass && t.Namespace == "Cinemachine").ToList();
```

Comment:
The reason for checking the namespace is to remove irrelevant classes in the list. It reduces the size by ~10.